apps/angular/projection/src/app/component/teacher-card/teacher-card.component.ts

```diff
@@ -1,40 +1,40 @@
+import { AsyncPipe } from '@angular/common';
 import { Component, OnInit } from '@angular/core';
-import { FakeHttpService } from '../../data-access/fake-http.service';
+import { randTeacher } from '../../data-access/fake-http.service';
 import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
 import { Teacher } from '../../model/teacher.model';
+import { DataCardComponentBase, DataStoreBase } from '../../shared';
 import { CardComponent } from '../../ui/card/card.component';
 
 @Component({
   selector: 'app-teacher-card',
   template: `
     <app-card
-      [list]="teachers"
-      [type]="cardType"
-      customClass="bg-light-red"></app-card>
+      [list]="data$ | async"
+      (addNewItem)="onAddNewItem()"
+      (deleteItem)="onDeleteItem($event)"
+      [backgroundColor]="'rgba(250, 0, 0, 0.1)'">
+      <img src="assets/img/teacher.png" image />
+    </app-card>
   `,
-  styles: [
-    `
-      ::ng-deep .bg-light-red {
-        background-color: rgba(250, 0, 0, 0.1);
-      }
-    `,
+  providers: [
+    {
+      provide: DataStoreBase,
+      useClass: TeacherStore,
+    },
   ],
   standalone: true,
-  imports: [CardComponent],
+  imports: [CardComponent, AsyncPipe],
 })
-export class TeacherCardComponent implements OnInit {
-  teachers: Teacher[] = [];
-  cardType = CardType.TEACHER;
-
-  constructor(
-    private http: FakeHttpService,
-    private store: TeacherStore,
-  ) {}
-
+export class TeacherCardComponent
+  extends DataCardComponentBase<Teacher>
+  implements OnInit
+{
   ngOnInit(): void {
     this.http.fetchTeachers$.subscribe((t) => this.store.addAll(t));
+  }
 
-    this.store.teachers$.subscribe((t) => (this.teachers = t));
+  override onAddNewItem() {
```

Comment:
OK, now I understand the “generic” naming.

From my experience, this adds another layer of indirection that does not provide a benefit for the developer.
This approach makes absolute sense to me if we are in the area of spinning up components dynamically from a configuration. Then we need to introduce contracts and hooks.

Reply:
The idea is to move all repeated functionality to abstract class (DRY), thus you don't need to copy paste it once you create a new card type component and focus only on the things that differs this component from others.

Reply:
> The idea is to move all repeated functionality to abstract class (DRY), thus you don't need to copy paste it once you create a new card type component and focus only on the things that differs this component from others.

Yes, I understand the goal. As far as I can tell, this approach only works in very few situations.
Often we tend to abstract too early. Then a business requirement changes, and we start adding conditions to our abstractions in order to make them work for a specific case.

That's why I value SRP (Single Responsibility Principle) over DRY.
In our challenge here, I would ask, what is a card responsible for? Aha, providing a visual boundary and a spot to render 
any content.